spec/tictactoe_spec.rb

```diff
@@ -7,7 +7,25 @@
   let (:test_ttt) { TicTacToe.new(game_io, player1, player2) }
 
   it "runs the game loop " do
-    allow(test_ttt.game_loop).to receive(:game_loop) { }
-    test_ttt.game_loop
+    expect(test_ttt.game_loop).to receive(:game_loop).once
+    expect(game_io).to receive(:puts_message)
+    expect(game_io).to receive(:get_input).and_return("N")
+    test_ttt.play_game
+  end
+
+  it "plays the game twice " do
+    expect(test_ttt.game_loop).to receive(:game_loop).twice
```

Comment:
I guess it is slightly confusing, as there is a GameLoop class that is a loop that plays the game. And the TicTacToe has a loop that calls that loop and then does a play again loop. So there are two separate game loops.